pkg/policy/l4.go

```diff
@@ -39,24 +40,36 @@ func (l7 L7DataMap) MarshalJSON() ([]byte, error) {
 		return []byte("[]"), nil
 	}
 
+	/* First, create a sorted slice of the selectors so we can get
+	 * consistent JSON output */
+	selectors := make(api.EndpointSelectorSlice, len(l7))
```

Comment:
Here you create a slice of *length* `len(l7)`. You should create an empty slice of *capacity* `len(l7)`:
```
selectors := make(api.EndpointSelectorSlice, 0, len(l7))
```

I'm surprised that the tests are passing...